Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -1003,6 +1003,21 @@ void RenderLayerCompositor::updateEventRegions()
     m_renderView.setNeedsEventRegionUpdateForNonCompositedFrame(false);
 }
 
+Vector<EventTrackingRegions> RenderLayerCompositor::touchEventRegionsForTesting() const
+{
+    Vector<EventTrackingRegions> eventRegions;
+    Vector<CheckedPtr<RenderLayer>> layers = { m_renderView.layer() };
+    while (!layers.isEmpty()) {
+        if (CheckedPtr layer = layers.takeLast()) {
+            eventRegions.append(layer->touchEventRegionsForTesting());
```

Comment:
This is still collecting a vector of regions in different coordinate systems.

I think the better approach here is to abandon `touchEventRegionsForTesting`, and instead have tests dump layer trees with the `LAYER_TREE_INCLUDES_EVENT_REGION` flag, and make sure your touch event regions show there.